src/util/SpinLock.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2021-present Samsung Electronics Co., Ltd
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+ *  USA
+ */
+
+#if defined(ENABLE_THREADING)
+
+#ifndef __EscargotSpinLock__
+#define __EscargotSpinLock__
+
+#include <atomic>
+
+namespace Escargot {
+
+class SpinLock {
```

Comment:
Q. As far as I know, spinlock causes performance degradation in multi-threaded. Is there any reason to introduce this?

Reply:
SpinLock is better than mutex if critical section is short.
because SpinLock doesn't use kernel resource(kernel system call)
AtomicOperations have short critical section, so I use SpinLock.

Reply:
Thanks for the explanation!